pkg/pdatatest/plogtest/options.go

```diff
@@ -0,0 +1,180 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plogtest // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/plogtest"
+
+import (
+	"bytes"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
+)
+
+// LogsCompareOption can be used to mutate expected and/or actual logs before comparing.
+type LogsCompareOption interface {
+	applyOnLogs(expected, actual plog.Logs)
+}
+
+// IgnoreResourceAttributeValue is a LogsCompareOption that removes a resource attribute
+// from all resources.
+func IgnoreResourceAttributeValue(attributeName string) LogsCompareOption {
+	return ignoreResourceAttributeValue{
+		attributeName: attributeName,
+	}
+}
+
+type ignoreResourceAttributeValue struct {
+	attributeName string
+}
+
+func (opt ignoreResourceAttributeValue) applyOnLogs(expected, actual plog.Logs) {
+	opt.maskLogsResourceAttributeValue(expected)
+	opt.maskLogsResourceAttributeValue(actual)
+}
+
+func (opt ignoreResourceAttributeValue) maskLogsResourceAttributeValue(metrics plog.Logs) {
+	rls := metrics.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		internal.MaskResourceAttributeValue(rls.At(i).Resource(), opt.attributeName)
+	}
+}
+
+func IgnoreObservedTimestamp() LogsCompareOption {
+	return ignoreObservedTimestamp{}
+}
+
+type ignoreObservedTimestamp struct{}
+
+func (opt ignoreObservedTimestamp) applyOnLogs(expected, actual plog.Logs) {
+	now := pcommon.NewTimestampFromTime(time.Now())
+	maskObservedTimestamp(expected, now)
+	maskObservedTimestamp(actual, now)
```

Comment:
You don't have to have a struct for every options, do something like:

```suggestion
func IgnoreObservedTimestamp() LogsCompareOption {
	return logsCompareOptionFunc(func(expected, actual plog.Logs) {
		now := pcommon.NewTimestampFromTime(time.Now())
		maskObservedTimestamp(expected, now)
		maskObservedTimestamp(actual, now)
	})
}

type logsCompareOptionFunc func(expected, actual plog.Logs)

func (f ignoreObservedTimestamp) applyOnLogs(expected, actual plog.Logs) {
	f(expected, actual)
}
```

Reply:
Yes, that was on my list of improvements just didn't have a chance to apply it. I can do it in this PR

Reply:
Updated